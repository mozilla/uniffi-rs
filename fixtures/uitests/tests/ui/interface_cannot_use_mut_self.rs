
// Unfortunately, path is relative to a temporary build directory :-/
uniffi_macros::generate_and_include_scaffolding!("../../../fixtures/uitests/src/counter.udl");

fn main() { /* empty main required by `trybuild` */}

pub struct Counter {
    value: u32,
}

impl Counter {
    pub fn new() -> Self {
        Self { value: 0 }
    }

    // This will fail to compile due to `&mut self` receiver.
    pub fn increment(&mut self) -> u32 {
        self.value = self.value + 1;
        self.value
    }
}

