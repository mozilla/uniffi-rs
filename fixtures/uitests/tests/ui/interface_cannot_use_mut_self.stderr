error[E0308]: mismatched types
   --> $DIR/counter.uniffi.rs:160:13
    |
160 |             &<std::sync::Arc<Counter> as uniffi::FfiConverter>::try_lift(ptr).unwrap(),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ types differ in mutability
    |
    = note: expected mutable reference `&mut Counter`
                       found reference `&Arc<Counter>`
